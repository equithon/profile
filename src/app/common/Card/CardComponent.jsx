import React from "react";
import styled from "styled-components";
import { Box } from "grommet";

const Container = styled(Box)`
  border-radius: ${props => props.theme.app.border.radius};
  color: ${props => props.theme.colors.offWhite};
  width: ${props => props.width};
  height: ${props => props.height};

  // only display cursor or darken on hover if non-static (interactive)
  cursor: ${props => (props.interactive ? "pointer" : "auto")};
  transition: filter 200ms ease-in-out;
  &:hover {
    filter: ${props =>
      props.interactive ? "brightness(95%)" : "brightness(100%)"};
  }
`;

const CardComponent = ({
  width,
  height,
  interactive,
  backgroundColor,
  backgroundImg,
  children
}) => (
  <Container
    width={width || "medium"}
    height={height}
    background={backgroundColor || `url(${backgroundImg})`}
    interactive={interactive}
  >
    {children}
  </Container>
);

export default CardComponent;
