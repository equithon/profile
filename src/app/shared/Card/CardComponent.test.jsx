/* eslint-disable no-undef */
import React from "react";

import Card from "./CardComponent";

describe("Button", () => {
  it("renders correctly", () => {
    const component = mountWithTheme(<Card />);
    expect(component).toMatchSnapshot();
  });
});
