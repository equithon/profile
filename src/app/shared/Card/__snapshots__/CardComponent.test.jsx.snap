// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  padding: 0.8em;
  border-radius: 6px;
  color: white;
  background-color: #454545;
  cursor: auto;
  will-change: transform;
  -webkit-transition: -webkit-transform 150ms ease-in-out;
  -webkit-transition: transform 150ms ease-in-out;
  transition: transform 150ms ease-in-out;
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
}

.c0:hover {
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
}

<_default>
  <styled.div
    backgroundColor="black"
    clickable={false}
    color="white"
  >
    <StyledComponent
      backgroundColor="black"
      clickable={false}
      color="white"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  padding: ",
              [Function],
              ";
  border-radius: ",
              [Function],
              ";

  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  background: ",
              [Function],
              ";

  // only display cursor or expand on hover if clickable
  cursor: ",
              [Function],
              ";

  will-change: transform;
  transition: transform 150ms ease-in-out;
  transform: scale(1.01);
  &:hover {
    transform: ",
              [Function],
              ";
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
        color="white"
      />
    </StyledComponent>
  </styled.div>
</_default>
`;
